Sources/Private/Model/DotLottie/DotLottieImageProvider.swift

```diff
@@ -38,12 +38,12 @@ class DotLottieImageProvider: AnimationImageProvider {
   let filepath: URL
 
   func imageForAsset(asset: ImageAsset) -> CGImage? {
-    imageCache.object(forKey: asset.name as NSString)
+    images[asset.name]
   }
 
   // MARK: Private
 
-  private var imageCache: NSCache<NSString, CGImage> = .init()
+  private var images = [String: CGImage]()
```

Comment:
If we store the `filepath.url` couldn't we recover from a memory pressure eviction? It seems risky to have a bunch of `CGImage`s in memory and not be able to release them on memory pressure warning as it may lead to apps being jetsam'd

Reply:
When loading a dot lottie file, we currently:
 1. decompress the zip archive to a temporary directory
 2. load the file contents into memory
 3. delete the files in the temporary directory

I think deleting the temporary unzipped files after loading them is a good idea because:
 1. I think we don't want to be leaving behind a large number of files in the host app's temporary directory. The OS cleans these up in theory but I'm not sure we should rely on that.
 2. Since the OS can clean up those files at any time in theory, I don't think its safe to assume that they will still exist at any point in the future

That means that we don't have a way to gracefully reload the images if they get removed from the `NSCache`, since the filepath URLs are no longer valid.

Reply:
I suppose we could re-decompress the zip archive to reload the images if necessary after they're evicted from the cache, but that functionality doesn't current exist.

Reply:
Gotcha, sounds good. Perhaps we could add a note about adding that behavior if needed?